### DISCUSSION TITLE
Extend FunctionDefinition by a function type id

### DISCUSSION THREAD

**Discussion by bossenti** [apache/streampipes#1143] [General]
While browsing through the API endpoint for SteamPipes functions I had the feeling that an additional attribute that identifies the type of a function could be valuable for the [`FunctionDefinition`](https://github.com/apache/streampipes/blob/cf33668a016921dc831ea90476fa07158dc91b7e/streampipes-model/src/main/java/org/apache/streampipes/model/function/FunctionDefinition.java#L27) class. This would allow a client to map existing functions in StreamPipes to the underlying implementation and to directly create an instance of the function from the API response. How do you like that @dominikriemer?
Does this even make sense?

**Comment by dominikriemer** [comment]
Hi @bossenti can you provide an example how using the type identifier would work?
Currently, the function is only registered in the core to enable monitoring, while the implementation purely exists on the extensions service.

**Reply by bossenti** [reply]
So lets say I implement a `FunctionA` that has `some-id` as `functionTypeId` and create an instance of this that registers itself in the core. My idea is that this information enables you to do something like `FunctionHandler.from(functionDefinition)` that uses the information provided by `functionDefinition` to instantiate a `FunctionA`. Thereby, `some-id` is used to get the mapping to the `FunctionA` class. Do you understand what my intention is? 🤓
